content/flare/consensus.md

```diff
@@ -287,3 +287,238 @@ Market forces cause quorum slices to overlap and this gives rise to the network-
+They operate in the presence of asynchrony and communication faults, which may cause different processes to receive messages in different orders, and they tolerate both crash and Byzantine process fault.
+The system is asynchronous in the sense that no bound can be placed on the time required for a computation or for a communication of a message.
+A crash fault causes a process to stop processing and generating messages.
+A Byzantine fault allows an arbitrary behavior by the faulty process, including generating of messages intended to mislead the other processes.
+
+The total order ensures consistency by having the same updates applied to the replicated pieces of data in the same order at all sites.
+
+Each broadcast message has a header consisting of a process identifier, a message sequence number and payload content.
+A typical implementation might use a digital signature to substantiate the source of a message (process identifier) for each node.
+Systems capable of tolerating Byzantine faults must be able to confirm the source of a message and to detect messages that have the same header but different contents and/or acknowledgements within the message.
+Another option is to also embed a digest of the message content in its identifier, so that acknowledgements of messages with different contents but same sequence number will be regarded as acknowledgements of distinct messages.
+The acknowledgement and re-transmission mechanisms can ensure that all versions of a message are delivered to all destinations.
+Digital signatures and digests are computationally expensive, but do not require the additional rounds of message exchange of alternative strategies.
+
+The output of the algorithms is a total order of messages that is identical at all non-faulty processes in the system.
+Message `x1` from process `a` follows a message `x0` from process `b` if `x1` acknowledges `x0` or `x1` acknowledges some other message `y` that follows `x0`, where `y` is not the same as `x0` and `y` is not the same as `x1`.
+This implies that the relation between messages is transitive and also each message follows itself, consequently the relation is also reflexive.
+A message `x1` from a non-Byzantine process `a` acknowledges a message `x0` only if the process has received all messages that `x0` follows before it broadcast `x1`.
+
+```text
+ a       b                chain
+ |       |                  |
+ |       |                  |
+ |       |                  |
+ |      `x0`------------->  |  ↓
+ |                          |
+ |     [`y`-------------->] |  ↓
+ |                          |
+`x1`--------------------->  |  ↓
+                            |
+                           ...
+ ```
+
+Consequently, a message from a non-Byzantine process follows all messages broadcast earlier from that process.
```

Comment:
Actually after thinking about it more, I think the explanation could be simplified. I don't know how we could represent this as a diagram without actually making it more confusing than it should be 😄 WDYT of:

```suggestion
The output of the algorithms is a total order of messages that is identical at all non-faulty processes in the system.

Messages from non-Byzantine processes follow all previously broadcast messages. The relation between messages is transitive, and since each message also follows itself, it is reflexive as well. Messages need to acknowledge previously broadcast messages to be considered non-faulty.
```

I think this is enough to convey the idea that messages need to acknowledge prior messages in order to be considered valid? WDYT?

Reply:
Looks good to me, uodated here https://github.com/optakt/optakt.github.io/pull/25/commits/7a07360c1ddc79a9eff45017b245bf1f89997096
Thanks!